helper/schema/resource_data.go

```diff
@@ -171,6 +172,14 @@ func (d *ResourceData) SetPartial(k string) {
 	}
 }
 
+func (d *ResourceData) MarkNewResource() {
+	d.isNew = true
+}
+
+func (d *ResourceData) IsNewResource() bool {
+	return d.isNew
+}
```

Comment:
So this is because if we have `func Create() { return Update() }`

then the Create func already sets id for the resource which means that `Id() == ""` in `Update()` function will never be true.

Resources with `ForceNew: true` are also covered:
https://github.com/TimeIncOSS/terraform/blob/f-schema-new-resource/helper/schema/resource.go#L119-L127

----- 

Looking at the code also makes me wonder why do we explicitly call `d.SetId("")` in most `Delete()` functions in resources, if we're resetting it here anyway. It seems superfluous.